backends/ebpf/runtime/ebpf_runtime.c

```diff
@@ -14,89 +14,192 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-/* The runtime emulates a loaded eBPF program and provides an interface to feed
-   packets to the emulated eBPF filter. */
-
-
-
-#include <string.h>
-#include <libgen.h>
-#define PCAP_DONT_INCLUDE_PCAP_BPF_H
-#include <pcap/pcap.h>
+/* The runtime emulates a loaded eBPF program. It parses a set of given
+  pcap files and executes a C program that processes packets extracted from
+  the capture files. The C program takes a packet structure (commonly an skbuf)
+  and returns an action value for each packet.
+ */
+
+#include <unistd.h>     // getopt()
+#include <ctype.h>      // isprint()
+#include <string.h>     // memcpy()
+#include <stdlib.h>     // malloc()
 #include "test.h"
+#ifdef CONTROL_PLANE
 #include "control.h"
-#include <unistd.h>
-#include <ctype.h>
+#endif
+#include "pcap_util.h"
+
 #define FILE_NAME_MAX 256
+#define PCAPIN  "_in.pcap"
+#define PCAPOUT "_out.pcap"
+#define DELIM   '_'
 
 static int debug = 0;
 
 void usage(char *name) {
-    fprintf(stderr, "This program parses a pcap file, "
-            "feeds the individual packets into a filter function, "
+    fprintf(stderr, "This program expects a pcap file pattern, "
+            "extracts all the packets out of the matched files"
+            "in the order given by the packet time,"
+            "then feeds the individual packets into a filter function, "
             "and returns the output.\n");
-    fprintf(stderr, "Usage: %s [-d] -f file.pcap\n", name);
+    fprintf(stderr, "Usage: %s [-d] -f file.pcap -n num_pcaps\n", name);
     fprintf(stderr, "Options:\n");
     fprintf(stderr, "\t-d: Turn on debug messages\n");
     fprintf(stderr, "\t-f: The input pcap file\n");
+    fprintf(stderr, "\t-n: Specifies the number of input interfaces\n");
```

Comment:
You should use consistent naming: -n interface_count if your error message is about interfaces.